bin_sort/bin_sort.c

```diff
@@ -0,0 +1,56 @@
+/* Following algorithm sorts the input array in ascending order
+ * Time Complexity : O(n)-> Avg case and O(n^2)-> Worst Case
+ */
+#include <stdio.h>
+#include <stdlib.h>
+
+struct bucket {
+	int count;
+	float* value;
+};
+
+int comparator(const float* first, const float* second) {
+	float a = *first, b = *second;
+	float result = a - b;
+	return (0 < result) - (result < 0);
+}
+
+void bucket_sort(int array[], int n) {
+	struct bucket buckets[3];
+	int a, b, c;
+	for (a = 0; a < 3; a++) {
+		buckets[a].count = 0;
+		buckets[a].value = malloc(sizeof(int) * n);
+	}
+	for (a = 0; a < n; a++) {
+		if (array[a] < 0) {
+			buckets[0].value[buckets[0].count++] = array[a];
+		}
+		else if (array[a] > 10) {
+			buckets[2].value[buckets[2].count++] = array[a];
+		}
+		else {
```

Comment:
Now that we have number in 0..1 range, all numbers are going in this bucket. 
So, is BinSort really effective here? 
See Java implementation.